### DISCUSSION TITLE
How to add LAN domain names to reverse proxy

### DISCUSSION THREAD

**Discussion by gabeklavans** [caprover/caprover#1785] [Q&A] [answered]
Hi, I have some apps running on the same pc as my Caprover instance. Currently I have to access these apps using the local IP of my home server and the port that the app is listening on (e.g. `192.168.88.x:xxxx`). I'd like to access these apps using a local domain name (e.g. app.local). How I set up the DNS for all that is one thing, but once I get that set up, I'm wondering how I should edit the nginx config to redirect my local domain name to an app running on the pc (e.g., direct `app.local` to `https://localhost:9443`).

I'm a bit of a noob when it comes to setting up reverse proxies, particularly with nginx, so any advice is appreciated.

**Comment by githubsaturn** [comment]
There is a one click app exactly for this purpose. It's "nginx-reverse-proxy".

You can set it up and use your port as the upstream address:
![image](https://github.com/caprover/caprover/assets/10582082/1e079336-a60c-4630-864f-99473f712872)

**Reply by gabeklavans** [reply]
Ah interesting, I actually just stumbled upon this one click app as you posted. So that would mean instead of making up `app.local` as the URL, it would be the CapRover-assigned `app-name.caprover-domain` URL. I suppose that's convenient enough for my purposes!

**Reply by gabeklavans** [reply]
> There is a one click app exactly for this purpose. It's "nginx-reverse-proxy".

So this actually was *not* what I wanted, as this actually makes my LAN application accessible to WAN using the caprover generated URL. I just want my LAN app to have a nicer URL.

What *does* achieve what I want, albeit in a bit of a clunky way, is the Nginx Redirect one-click-app, since this basically just does a full URL substitution. I can redirect to a LAN address, which is useless via WAN.

**Reply by githubsaturn** [reply]
ah I didn't realize that your CapRover is accessible from WAN. Yes, in that case, the only way to do is to redirect. Otherwise, anything else that you do in CapRover is accessible from outside.

**Reply by gabeklavans** [reply]
My nested question still stands: is there a way for me to add my own entries to the nginx config that caprover manages? So I can set up the reverse proxy myself based on a `*.local` domain name? I tried editing one of the configs exposed via the nginx override in the settings page, but I didn't quite seem to get a working result from that. That could just be my bad, but it's hard for me to evaluate if it's even possible, currently.

**Reply by githubsaturn** [reply, marked as answer]
> is there a way for me to add my own entries to the nginx config that caprover manages?

Yes  you can. Navigate to settings > edit nginx > "CapRover Config Location in nginx container"

You can add any custom `server {}` block at the very top. Make sure to restart your service for this to take effect.

**Reply by gabeklavans** [reply]
Great, I'll give that a shot later! Just for the record, I did find that setting from your docs. I tried to set it up and it seemed to be failing but that very well could have been my fault, so I wanted to see if there was a better way. Thanks for the initial suggestion with the one click app, in any case.

**Reply by gabeklavans** [reply]
With a proper nginx server block added to the CapRover config and an mDNS broadcast running to get the local domain up, I was able to achieve exactly what I wanted :) Thanks!

I did actually mess up the server block on the first try with a missing `;` and had to refer to https://github.com/caprover/caprover/issues/412#issuecomment-484077130 to help me get my instance back up and running... fun times!

**Reply by gabeklavans** [reply]
> at the very top

Oh just FYI, putting my server blocks at the very top actually bricked my caprover, as well. I had to put them all at the very bottom for it to work.

**Reply by githubsaturn** [reply]
There must be something else causing it. The ordering wouldn't matter in this case.

**Reply by gabeklavans** [reply]
Huh, weird. Maybe I missed a closing brace or something, but I thought I only changed the order and it worked. All well
